python/kotekan/hfbbuffer.py

```diff
@@ -0,0 +1,779 @@
+"""Read a visBuffer dump into python.
+"""
+# === Start Python 2/3 compatibility
+from __future__ import absolute_import, division, print_function, unicode_literals
+from future.builtins import *  # noqa  pylint: disable=W0401, W0614
+from future.builtins.disabled import *  # noqa  pylint: disable=W0401, W0614
+
```

Comment:
Generally I would suggest figuring out how to merge as much common functionality between `visbuffer.py` and `hfbbuffer.py` as it looks like there's a lot of identical stuff.

Reply:
I've moved the common stuff into `timespec.py`.

Reply:
It's not just the timespec stuff I'm referring to in here. On the more trivial end there's another random function that is copied (freq_id_to_stream_id at the very end of the file), but I'm primarily thinking that the bulk of `HFBRaw` is the same as `VisRaw`, and as that's basically determined by the fact that the same writer code is used that writes metadata + data blocks into a 2D time-freq array it could be factored out into a common base class. The near term advantage of that is less code to maintain and fix, at the expense of more work now, the long term gain is that supporting new data formats that fit this time-freq pattern becomes super easy.

It might be that refactoring that is too much work for the moment, but it would be great if you could look over that and see how possible it is, and how long it might take, so we can see if we want to defer that or not.